### PR TITLE
fix: implement less flaky network module

### PR DESCRIPTION
**Pull request by Lightning00Blade**
The `NetworkStorage` is the entry point of all request we store both `requests` and `intecepts` in it.
Each CDP event should try to get get a request and if not create it, that way the EventLifecycle of the Request is handle only by the `NetworkRequest` class.
Now Network events are non-blocking (we may want to revert part of this if we encounter flakiness).
To achieve this we have heuristics that determine when an event should be emitted.
These heuristics are backed up by Unit tests so that it's have easier way of reproducing different behavior when it comes to how events are ordered or if emitted at all from CDP.

### CODE REVIEW

File: tests/network/test_continue_response.py [hunk 1]
```diff
@@ -262,11 +256,12 @@ async def test_continue_response_must_specify_both_status_and_headers(
 
 
 @pytest.mark.asyncio
-@pytest.mark.skip(reason="TODO: Fix this test")
+@pytest.mark.skip(reason="CDP does not update the response correctly")
```
Comment (Lightning00Blade) [inline tests/network/test_continue_response.py:259]: Let's investigate separately.

File: tests/network/test_provide_response.py [hunk 2]
```diff
@@ -209,10 +204,10 @@ async def test_provide_response_non_blocked_request(websocket, context_id,
 
 
 @pytest.mark.asyncio
-@pytest.mark.skip(reason="TODO: Fix this test")
+@pytest.mark.skip(reason="CDP does not update the response correctly")
```
Comment (Lightning00Blade) [inline tests/network/test_provide_response.py:207]: Let's investigate separately.

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 3]
```diff
@@ -108,27 +115,26 @@ export class NetworkRequest {
     return this.#fetchId;
   }
 
-  get phase(): Network.InterceptPhase | undefined {
+  /**
+   * When block returns the phase for it
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkRequest.ts:119]:
```suggestion
   * When blocked returns the phase for it
```

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 4]
```diff
@@ -108,27 +115,26 @@ export class NetworkRequest {
...
+   */
+  get blockedIn(): Network.InterceptPhase | undefined {
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkRequest.ts:121]:
```suggestion
  get currentInterceptPhase(): Network.InterceptPhase | undefined {
```

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 5]
```diff
@@ -137,13 +143,15 @@ export class NetworkRequest {
     return Boolean(this.#request.info);
   }
 
-  #isBlockedInPhase(phase?: Network.InterceptPhase) {
-    return this.#networkStorage.requestBlockedBy(this, phase).size > 0;
+  #isBlockedByInPhase(phase: Network.InterceptPhase) {
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkRequest.ts:146]: Is blocked is confusing I think? Let's check if there are interceptors for the phase instead so renaming `isInterceptedInPhase`

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 6]
```diff
@@ -137,13 +143,15 @@ export class NetworkRequest {
...
+    return this.#networkStorage.requestBlockedBy(this, phase);
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkRequest.ts:147]: let's also rename the method in storage to be something like getInterceptsForRequestPhase

File: wpt-metadata/mapper/headless/webdriver/tests/bidi/network/add_intercept/url_patterns.py.ini [hunk 7]
```diff
@@ -1,117 +1,2 @@
...
-
-  [test_string_patterns_matching[https://{wpt_host}:443/-https://{wpt_host}:443/\]]
-    expected: FAIL
-
-  [test_string_patterns_matching[https://{wpt_host}/#ref-https://{wpt_host}/\]]
-    expected: FAIL
-
-  [test_string_patterns_matching[https://{wpt_host}/#ref1-https://{wpt_host}/#ref2\]]
-    expected: FAIL
+  expected: TIMEOUT
```
Comment (OrKoN) [inline wpt-metadata/mapper/headless/webdriver/tests/bidi/network/add_intercept/url_patterns.py.ini:2]: What causes timeouts?

**Comment by OrKoN** [comment]
Looks like perform_actions/invalid.py times out consistently

File: wpt-metadata/mapper/headful/webdriver/tests/bidi/network/add_intercept/url_patterns.py.ini [hunk 8]
```diff
@@ -1,117 +1,2 @@
...
-
-  [test_string_patterns_matching[https://{wpt_host}:443/-https://{wpt_host}:443/\]]
-    expected: FAIL
-
-  [test_string_patterns_matching[https://{wpt_host}/#ref-https://{wpt_host}/\]]
-    expected: FAIL
-
-  [test_string_patterns_matching[https://{wpt_host}/#ref1-https://{wpt_host}/#ref2\]]
-    expected: FAIL
+  expected: TIMEOUT
```
Comment (OrKoN) [inline wpt-metadata/mapper/headful/webdriver/tests/bidi/network/add_intercept/url_patterns.py.ini:2]: Still times out?

**Review by OrKoN — APPROVED** [review]

**Comment by OrKoN** [comment]
@Lightning00Blade please update the description with the summary of the fix

**Review by OrKoN — APPROVED** [review]

**Pull request merged**